### PR TITLE
many: add `snap download` command that downloads into /var/lib/snapd/download

### PR DESCRIPTION
**Pull request by mvo5**
This is an initial and minimal implementation for `snap download`. It will download the given snap to  `/var/lib/snapd/download`. This is enough for me to continue with the spread image test plan (that involves downloading and modifying the ubuntu-core image. Some unit tests missing because I want to check the direction.

However its probably not yet good enough for end-user consumption. I'm keen to hear what we should support. Client selected output dir for download? Client selected target file? Json output so that it can be controlled from scripts? Or YAGNI and add all this once we have use-cases?

### CODE REVIEW

File: overlord/snapstate/snapmgr.go [hunk 1]
```diff
@@ -509,7 +510,17 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	ss.SnapPath = downloadedSnapFile
+	// used by snapstate.Downlaod()
+	if ss.SnapPath != "" {
```
Comment (chipaca) [inline overlord/snapstate/snapmgr.go]: could you invert the logic here? I find it sometimes hard to follow `if (negation) {} else {}`

**Comment by chipaca** [comment]
:+1:

**Comment by mvo5** [comment]
Closing for now as Gustavo and Samuelle suggested a different approach. The snap should be downloaded into the current directory without tasks which makes it useful for prepare-image and we can share code here.
